pkg/nodes.go

```diff
@@ -0,0 +1,88 @@
+package pkg
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/go-logr/logr"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/openshift-kni/node-label-operator/api/v1beta1"
+)
+
+// RemoveOwnedLabels removes all uncovered owned labels from the node and return true if the node was modified
+func RemoveOwnedLabels(node *v1.Node, allOwnedLabels []v1beta1.OwnedLabels, allLabels []v1beta1.Labels, log logr.Logger) bool {
+	// check if we have owned labels on the node
+	log.Info("Checking owned labels", "node", node.Name)
+	nodeModified := false
+	for labelDomainName := range node.Labels {
+		// check if we own this label
+		for _, ownedLabel := range allOwnedLabels {
+			if !IsOwnedLabel(labelDomainName, ownedLabel, log) {
+				continue
+			}
+			// we own this label
+			// check if it is still covered by a label rule
+			if !IsCoveredByAll(node.Name, labelDomainName, allLabels, log) {
+				// we need to remove the label
+				log.Info("Deleting uncovered owned label")
+				nodeLabels := node.Labels
+				delete(nodeLabels, labelDomainName)
+				node.Labels = nodeLabels
```

Comment:
This line is redundant, as they point to the same map.
Also the 3 lines could be reduced to:
`delete(node.Labels, labelDomainName)`

Reply:
yep, thx